src/rhsmlib/facts/firmware_info.py

```diff
@@ -130,7 +130,8 @@ def get_firmware_collector(arch, prefix=None, testing=None,
         try:
             import dmidecode  # noqa
             firmware_provider_class = dmiinfo.DmiFirmwareInfoCollector
-        except:
+        except ImportError:
+            log.debug("Firmware info collector is not available.")
```

Comment:
should we print the import error message? ie something like
```python
        except ImportError as exc:
            log.debug(f"cannot import dmidecode: {exc}")
```

Reply:
I wasn't sure which message would fit better here -- if general "not available", or specific "can't import" one. Both would work, it depends on who the person reading the logs will be and what they'll be looking for. Maybe we could do both with
```py
        except ImportError as exc:
            log.debug("Firmware info collector is not available: {exc}.")
```

Reply:
It's better to print `dmidecode` than `Firmware info collector`. We sub-man developers usually read the log containing debug messages.

Reply:
Done.